File: practica parte 2/Ejercicio 1: Empleado y ArchivoEmpleado/src/empleado.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

class Empleado implements Serializable {
    private String nombre;
    private float sueldo;
    
    public Empleado(String nombre, float sueldo) {
        this.nombre = nombre;
        this.sueldo = sueldo;
    }
    
    public String getNombre() {
        return nombre;
    }
    
    public float getSueldo() {
        return sueldo;
    }
}

class ArchivoEmpleado {
    private String nomA;
    
    public ArchivoEmpleado(String nombreArchivo) {
        this.nomA = nombreArchivo;
    }
    
    public void crearArchivo() {
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nomA))) {
            oos.writeObject(new ArrayList<Empleado>());
        } catch (IOException e) {
            System.out.println("Error al crear archivo: " + e.getMessage());
        }
    }
    
    public void guardarEmpleado(Empleado e) {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nomA))) {
            List<Empleado> empleados = (List<Empleado>) ois.readObject();
            empleados.add(e);
            
            try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nomA))) {
                oos.writeObject(empleados);
            }
        } catch (IOException | ClassNotFoundException ex) {
            System.out.println("Error al guardar empleado: " + ex.getMessage());
        }
    }
    
    public Empleado buscaEmpleado(String n) {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nomA))) {
            List<Empleado> empleados = (List<Empleado>) ois.readObject();
            
            for (Empleado emp : empleados) {
                if (emp.getNombre().equals(n)) {
                    return emp;
                }
            }
        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error al buscar empleado: " + e.getMessage());
        }
        return null;
    }
    
    public Empleado mayorSalario(float sueldo) {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nomA))) {
            List<Empleado> empleados = (List<Empleado>) ois.readObject();
            
            for (Empleado emp : empleados) {
                if (emp.getSueldo() > sueldo) {
                    return emp;
                }
            }
        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error al buscar empleado: " + e.getMessage());
        }
        return null;
    }

    public static void main(String[] args) {
        ArchivoEmpleado archivo = new ArchivoEmpleado("empleados.dat");
        archivo.crearArchivo();
        
        archivo.guardarEmpleado(new Empleado("Juana Garcia", 2500));
        archivo.guardarEmpleado(new Empleado("Maria Chavez", 3500));
        
        Empleado emp = archivo.buscaEmpleado("Juana Garcia");
        if (emp != null) {
            System.out.println("Empleado encontrado: " + emp.getNombre() + ", Sueldo: " + emp.getSueldo());
        }
        
        Empleado empMayor = archivo.mayorSalario(3000);
        if (empMayor != null) {
            System.out.println("Empleado con sueldo mayor: " + empMayor.getNombre());
        }
    }
}

File: practica parte 2/Ejercicio 1: Empleado y ArchivoEmpleado/empleado.py
import pickle

class Empleado:
    def __init__(self, nombre, sueldo):
        self.nombre = nombre
        self.sueldo = sueldo
    
    def get_nombre(self):
        return self.nombre
    
    def get_sueldo(self):
        return self.sueldo

class ArchivoEmpleado:
    def __init__(self, nombre_archivo):
        self.nomA = nombre_archivo
    
    def crear_archivo(self):
        try:
            with open(self.nomA, 'wb') as f:
                pickle.dump([], f)
        except IOError:
            print("Error al crear archivo")
    
    def guardar_empleado(self, empleado):
        try:
            with open(self.nomA, 'rb') as f:
                empleados = pickle.load(f)
            
            empleados.append(empleado)
            
            with open(self.nomA, 'wb') as f:
                pickle.dump(empleados, f)
        except IOError:
            print("Error al guardar empleado")
    
    def busca_empleado(self, nombre):
        try:
            with open(self.nomA, 'rb') as f:
                empleados = pickle.load(f)
            
            for emp in empleados:
                if emp.get_nombre() == nombre:
                    return emp
            return None
        except IOError:
            print("Error al buscar empleado")
            return None
    
    def mayor_salario(self, sueldo):
        try:
            with open(self.nomA, 'rb') as f:
                empleados = pickle.load(f)
            
            for emp in empleados:
                if emp.get_sueldo() > sueldo:
                    return emp
            return None
        except IOError:
            print("Error al buscar empleado")
            return None

archivo = ArchivoEmpleado("empleados.dat")
archivo.crear_archivo()

archivo.guardar_empleado(Empleado("Juana Garcia", 2500))
archivo.guardar_empleado(Empleado("Maria Chavez", 3500))

emp = archivo.busca_empleado("Juana Garcia")
if emp:
    print(f"Empleado encontrado: {emp.get_nombre()}, Sueldo: {emp.get_sueldo()}")

emp_mayor = archivo.mayor_salario(3000)
if emp_mayor:
    print(f"Empleado con sueldo mayor: {emp_mayor.get_nombre()}")